src/components/AddItem.jsx

```diff
@@ -24,37 +25,73 @@ function AddItem() {
   const [itemName, setItemName] = useState('');
   const [frequency, setFrequency] = useState(7);
   const [notification, setNotification] = useState('');
+  const [list, setList] = useState([]);
+  const localToken = localStorage.getItem('list-token');
+  const [duplicateMessage, setDuplicateMessage] = useState(null);
 
-  //retrive the token from localStorage
-  const token = localStorage.getItem('list-token');
+  //useEffect to setList of items in that user's list which will be used for duplicate comparison
+  useEffect(() => {
+    const unsubscribe = onSnapshot(collection(db, localToken), (snapshot) => {
+      const snapshotDocs = [];
+      snapshot.forEach((doc) => snapshotDocs.push(doc.data()));
+      setList(snapshotDocs);
+    });
+    return () => {
+      unsubscribe();
+    };
+  }, [localToken]);
+
+  function duplicateCheck(itemName, list) {
+    let isDuplicateFound = false;
+    list.forEach((listItem) => {
+      listItem.itemName = listItem.itemName.toLowerCase();
+      itemName = itemName.toLowerCase();
+      //regex for removing punctuation and spaces from firebase item and form input item
+      listItem.itemName = listItem.itemName.replace(
+        /[.,\/#!$%\^&\*;:{}=\-_`~()]/g,
+        '',
+      );
+      listItem.itemName = listItem.itemName.replace(/\s{2,}/g, '');
+      itemName = itemName.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g, '');
+      itemName = itemName.replace(/\s{2,}/g, '');
+      if (listItem.itemName === itemName) {
+        return (isDuplicateFound = true);
+      }
+    });
+    return isDuplicateFound;
+  }
 
-  //handle submit for the form
   const handleSubmit = async (e) => {
     e.preventDefault();
-    //create new doc reference for text input and save to firestore database
-    //use a token to save new items into specific collection (shopping list)
+    let isDuplicateFound = duplicateCheck(itemName, list);
+    if (isDuplicateFound) {
+      // sets Error message if duplicateCheck results in isDuplicateFound === true
+      // if isDuplicateFound returns, preventing item from being written to db
+      setDuplicateMessage(`${itemName} already exists in your list!`);
+      return;
+    } else {
```

Comment:
we don't need continue with else here as an duplicateMessage already set to null